src/components/crownConfig.vue

```diff
@@ -143,6 +148,9 @@ export default {
         diagram: this.moddle.create('bpmndi:BPMNEdge'),
       });
     },
+    remove() {
+      return this.isLane ? this.removePoolLaneShape() : this.removeShape();
```

Comment:
This logic is in a mixin. Maybe there is a better way.

Reply:
I think for this the inheritance seems to be reversed. Currently `crownConfig` extends `poolLaneCrownConfig`, which means crown config has to know about pool lanes. Would it be possible to have `poolLaneCrownConfig` extend `crownConfig` instead? It would then be possible to have a single `remove()` method on `crownConfig` which `poolLaneCrownConfig` can then override to add the specific behaviour. This would also mean that a generic component doesn't need to know anything about the specific behaviour of the pool lane.

Reply:
It doesn't seem to work in that order. Or there are things in the crownConfig I need to make sure run first.

Reply:
Can you be more specific? Have you already tried this? What things need to run first? What errors did you see?

Reply:
Think I got it. Thanks for the suggestion.

Reply:
In the crownConfig it updates things like plane elements, artifacts, and flow elements.

I may have an issue to solve. Elements aren't embedding in the pool.